File: sdk/typescript/src/types/embedding.ts
import type { Usage, AgentCCMetadata } from "./shared.js";

export interface Embedding {
  object: "embedding";
  index: number;
  embedding: number[] | string;
}

export interface EmbeddingResponse {
  object: "list";
  data: Embedding[];
  model: string;
  usage: Usage;
  agentcc?: AgentCCMetadata;
  [key: string]: unknown;
}

export interface EmbeddingCreateParams {
  model: string;
  input: string | string[] | number[] | number[][];
  encoding_format?: "float" | "base64";
  dimensions?: number;
  user?: string;
  extra_headers?: Record<string, string>;
  [key: string]: unknown;
}

File: sdk/python/src/agentcc/types/embedding.py
"""Embedding response types."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict

from agentcc.types.agentcc_metadata import AgentCCMetadata
from agentcc.types.shared import Usage


class Embedding(BaseModel):
    object: str = "embedding"
    embedding: list[float]
    index: int
    model_config = ConfigDict(extra="allow")


class EmbeddingResponse(BaseModel):
    object: str = "list"
    data: list[Embedding]
    model: str
    usage: Usage | None = None
    agentcc: AgentCCMetadata | None = None
    model_config = ConfigDict(extra="allow")
